File: Practice/src/coding_basic_training/MakeLengthOfArrayPowerOf2.js
function solution(arr) {
    var answer = Array().concat(
        ...arr,
        Array(
            Math.pow(2, (arr.length - 1).toString(2).length) % arr.length
        ).fill(0)
    );
    // return [
    //     ...arr,
    //     ...Array(2 ** Math.ceil(Math.log2(arr.length)) - arr.length).fill(0),
    // ];
}

File: Practice/src/coding_basic_training/MakeLengthOfArrayPowerOf2.java
package coding_basic_training;

import java.util.ArrayList;

public class MakeLengthOfArrayPowerOf2 {
//	배열의 길이를 2의 거듭제곱으로 만들기
	public int[] solution(int[] arr) {
		ArrayList<Integer> li = new ArrayList<Integer>();
		for (int i : arr) {
			li.add(i);
		}
		for (int i = 1;; i *= 2) {
			if (i >= li.size()) {
				for (int j = li.size(); j < i; j++) {
					li.add(0);
				}
				break;
			}
		}
		return li.stream().mapToInt(i -> i).toArray();
//        return Arrays.copyOf(arr, (int) Math.pow(2, Math.ceil(Math.log(arr.length) / Math.log(2))));
	}
}